### ISSUE TITLE
Table transform to HTML does not satisfy Web Content Accessibility Guidelines

### ISSUE THREAD

**Issue by MelHopper** [mkdocs/mkdocs#2323]
**product level**
mkdocs-material==4.5.0
mkdocs-minify-plugin==0.2.1  # via mkdocs-material
mkdocs==1.0.4

The same issue seems to still appear in the following levels:

mkdocs-material ==7.0.3
mkdocs ==1.1.2
mkdocs-material-extensions==1.0.1

**Summary**

The transform from markdown to HTML tables adds basic tags <th>, <td> etc, but this is not enough to pass AA compliance of Web Content Accessibility Guidelines (WCAG).

Without adding extra tags, users cannot determine where they are in the table as the screenreader reads the cell contents. see the following for more info:
- The IBM Accessibility toolkit gives AA compliance of WRAG 2.1 at Level 3. The updates to the mkdocs table transform would satisfy level 2 requirements so is a basic acccessibility need. https://www.ibm.com/able/toolkit/design/content/text-meaning
-  https://a11y-101.com/development/tablesfor more info

Legal obligations to use the WCAG 2.1 guidelines are detailed [here] (https://en.wikipedia.org/wiki/Web_Content_Accessibility_Guidelines) by country.

**Details**

To be accessible to screenreaders, HTML tables must also include one of the following:

- use `scope=col` and `scope=row' to the cells to indicate which ones are column headers and which are row headers
or
- use a `headers` attribute to every data cell referencing the ids of the headers for that cell. 

https://www.ibm.com/able/toolkit/design/content/text-meaning/

For example:

The following markdown would transformed into the following html when appropriate tagging is added:

|Fruit  | Price | weight
| :------------ | :-----------: | :-----------: |
| Apple | 65p      | 70g |
| Pear   | 73p      | 55g |

Becomes:

```xml

<table style="width:100%">

  <tr>
    <th scope="col">Fruit</th>
    <th scope="col">Price</th>
    <th scope="col">Weight</th>
  </tr>

  <tr>
    <td scope="row"> Apple</td>
    <td 65p</td>
    <td 70g</td>
  </tr>

  <tr>
    <td scope="row"> Pear</td>
    <td 73p</td>
    <td 55g</td>
  <tr>
</table>
```

**Comment by waylan** [comment]
All Markdown processing is handled by [Python-Markdown][1]. Any changes should be addressed upstream. That said, the existing behavior of tables is common across multiple Markdown implementations and I don't see it being changed unless that change happens across most Markdown implementations which include support.

Regardless, you are always free to create your own Markdown table extension which meets your needs.

[1]: https://github.com/Python-Markdown/markdown

**Comment by squidfunk** [comment]
On a side note, `scope` is not part of HTML 5:
https://www.tutorialspoint.com/html5/html5_deprecated_tags.htm

**Comment by waylan** [comment]
For the record, I think some of this could probably be addressed by Markdown. However, there are a few sticking points. However, as previously stated, that is a discussion which needs to happen in the Python-Markdown project not here. If this is important to you, please follow up by opening an issue there.

**Comment by waylan** [comment]
I think some of this could probably be addressed by Markdown. However, there are a few sticking points. However, as previously stated, that discussion needs to happen in th ePython-Markdown protract If this is important to you, please open an issue there.
